### PR TITLE
[REVIEW] fix warning unreachable code in datetime.cuh

### PR DESCRIPTION
**Pull request by karthikeyann**
related to https://github.com/rapidsai/cudf/issues/5272

fixes warning of unreacheable code in datetime.cuh

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by davidwendt — APPROVED** [review]

File: cpp/src/io/csv/datetime.cuh [hunk 1]
```diff
@@ -435,34 +435,36 @@ __inline__ __device__ int64_t parseTimeDeltaFormat(const char *data, long start,
...
-      1L, 10L, 100L, 1000L, 10000L, 100000L, 1000000L, 10000000L, 100000000L, 1000000000L};
-    nanosecond *= powers_of_ten[9 - num_digits];
+    //:%M:%S
+    if (data[moving_pos] == sep) { minute = parse_integer<int>(data, ++moving_pos, end); }
+    if (data[moving_pos] == sep) { second = parse_integer<int>(data, ++moving_pos, end); }
+    if (std::is_same<T, cudf::duration_s>::value) {
+      return ((days * 24L + hour) * 60L + minute) * 60L + second;
+    }
+    //.n
+    if (data[moving_pos] == '.') {
```
Comment (harrism) [inline cpp/src/io/csv/datetime.cuh:460]: else if?

**Review by harrism — APPROVED** [review]

File: cpp/src/io/csv/datetime.cuh (hunk 1 above)
Comment (karthikeyann) [inline cpp/src/io/csv/datetime.cuh:460]:
```suggestion
    } else if (data[moving_pos] == '.') { //.n
```

File: cpp/src/io/csv/datetime.cuh [hunk 2]
```diff
@@ -435,34 +435,34 @@ __inline__ __device__ int64_t parseTimeDeltaFormat(const char *data, long start,
...
-    int8_t num_digits                 = min(9L, moving_pos - start_subsecond);
-    constexpr int64_t powers_of_ten[] = {
-      1L, 10L, 100L, 1000L, 10000L, 100000L, 1000000L, 10000000L, 100000000L, 1000000000L};
-    nanosecond *= powers_of_ten[9 - num_digits];
+    //:%M:%S
+    if (data[moving_pos] == sep) { minute = parse_integer<int>(data, ++moving_pos, end); }
+    if (data[moving_pos] == sep) { second = parse_integer<int>(data, ++moving_pos, end); }
+    if (std::is_same<T, cudf::duration_s>::value) {
+      return ((days * 24L + hour) * 60L + minute) * 60L + second;
+    } else if (data[moving_pos] == '.') { //.n
```
Comment (karthikeyann) [inline cpp/src/io/csv/datetime.cuh:458]:
```suggestion
    } else if (data[moving_pos] == '.') {  //.n
```

**Comment by galipremsagar** [comment]
rerun tests

**Pull request merged**